model/spectrum.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * spectrum.h
+ */
+#ifndef DXTBX_MODEL_SPECTRUM_H
+#define DXTBX_MODEL_SPECTRUM_H
+
+#include <iostream>
+#include <cmath>
+#include <scitbx/vec3.h>
+#include <scitbx/array_family/shared.h>
+#include <scitbx/array_family/simple_io.h>
+#include <scitbx/array_family/simple_tiny_io.h>
+#include <dxtbx/error.h>
+#include "model_helpers.h"
+
+namespace dxtbx { namespace model {
+  typedef scitbx::af::shared<double> vecd;
+
+  /** A class to represent a 2D spectrum. */
+  class Spectrum {
+  public:
+    /** Default constructor: initialise all to zero */
+    Spectrum() {}
+
+    /**
+     * Initialise all the spectrum parameters.
+     * @param energies The spectrum energies (eV)
+     * @param energies The spectrum weights (unitless)
+     */
+    Spectrum(vecd energies, vecd weights)
```

Comment:
Strikes me while defining this as a type that adding convenience methods to give minimum / maximum energy for prediction purposes could be helpful i.e. where the weights are measurably above 0 (I appreciate that there is some judgement here) 

Motivation: for purposes of computing predictions means we have a window to predict through, e.g. if the spectrum in question is very narrow, rather than `min(energies) -> max(energies)` which could be a lot wider.

Reply:
I suppose I'd need an algorithm laid out before I implemented something?

Reply:
OK, algorithm - 
- compute CDF from spectrum
- identify the bin below the 1% point
- identify the bin above the 99% point

There you go - a pretty plausible min, max window with nothing more than floating adds and comparisons (i.e. super cheap in C++)

If we add a Gaussian constructor can use the same logic to define 3σ window or something, but gives a consistent and reasonable definition.

Reply:
OK, so maybe this is not so simple - I have just grabbed a random SwissFEL Jungfrau data set I had lying around and it looks like the spectrum is not background subtracted i.e.:

![146-00001](https://user-images.githubusercontent.com/1241716/88805699-61df7700-d1a7-11ea-8ef0-d8a63c6d49de.png)

Without this will be kinda game over for my suggestions, but I'll keep looking for a bit. Am trying to get to this now.